### DISCUSSION TITLE
Use 'required' members when targeting net6.0;net7.0

### DISCUSSION THREAD

**Discussion by tmds** [dotnet/csharplang#6797] [Q&A] [answered]
I'd like to use _required members_ for a project that targets both `net6.0` and `net7.0`.

The .NET 6 build gives build errors for missing attributes that were introduced in .NET 7 to support _required members_:

```
error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.RequiredMemberAttribute..ctor'
error CS0656: Missing compiler required member 'System.Runtime.CompilerServices.CompilerFeatureRequiredAttribute..ctor'
error CS0656: Missing compiler required member 'System.Diagnostics.CodeAnalysis.SetsRequiredMembersAttribute..ctor'
```

Can I add some package that defines these attributes, or define them myself to make the build work?

**Comment by HaloFour** [comment]
Officially, C# 11 is only supported on .NET 7.0 and newer.  Microsoft doesn't provide assemblies to polyfill older runtimes.

See: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/configure-language-version

You can provide your own, though, and there may be third-party NuGet packages out there to do the same.

**Comment by Sergio0694** [comment, marked as answer]
> "Can I add some package that defines these attributes"

Sure you can! Just add [PolySharp](https://github.com/Sergio0694/PolySharp) and enable C# 11 😄

> **Note**: as other mentioned, this scenario is not officially supported.

**Reply by tmds** [reply]
Great, thank you!

**Comment by pinkfloydx33** [comment]
I actually found myself in a similar situation... I really didn't care about net6.0, but wanted the net7.0 build to leverage `required` while I still need to multi-target my internal libraries. Only way I found around it was some nasty `#ifdefs`, enough so that I just gave up. I realize its not supported and I get that...but it'd be nice if it could somehow just be ignored

**Reply by saint4eva** [reply]
https://github.com/Sergio0694/PolySharp
